KomaMRIBase/src/timing/TrapezoidalIntegration.jl

```diff
@@ -28,6 +28,10 @@ function trapz(Δt::AbstractVector{T}, x::AbstractVector{TX}) where {T<:Real, TX
     y = sum(y)
     return y
 end
+@inline function trapz!(ϕ::AbstractVector{T}, Tz::AbstractMatrix{TX}, Δt::AbstractMatrix{T}, x::AbstractMatrix{TX}) where {T<:Real, TX<:Union{T, Complex{T}}}
+    Tz .= (x[:, 2:end] .+ x[:, 1:end-1]) .* Δt .* T(-π .* γ)
+    ϕ .= sum(Tz, dims=2)
+end
```

Comment:
I looked into changing how the ADC indexing works but wasn't able to find a faster / simpler way of doing it than how it is now. We want to hold onto the last column of the phi cumsum since we need it to compute M.xy, but we also want to use the in-place cumsum!, as it is slightly faster than phi .= cumsum(delta_phi, dims=2). And if possible, we want to avoid allocating additional memory for phi_ADC.

Reply:
No problem! it looks good